pydantic/main.py

```diff
@@ -71,9 +73,20 @@ def __new__(
     ) -> type:
         if _base_class_defined:
             class_vars: set[str] = set()
+            base_private_attributes: dict[str, ModelPrivateAttr] = {}
```

Comment:
can we move this into a separate function?

Reply:
no worries if it doesn't make sense.

Reply:
sure, does this seem good to you?
```python
def _collect_bases_data(bases) -> tuple[set[str], set[str], dict[str, ModelPrivateAttr]]:
    field_names: set[str] = set()
    class_vars: set[str] = set()
    private_attributes: dict[str, ModelPrivateAttr] = {}
    for base in bases:
        if _base_class_defined and issubclass(base, BaseModel) and base != BaseModel:
            # model_fields might not be defined yet in the case of generics, so we use getattr here:
            field_names.update(getattr(base, 'model_fields', {}).keys())
            class_vars.update(base.__class_vars__)
            private_attributes.update(base.__private_attributes__)
    return field_names, class_vars, private_attributes
```

that changes this to:
```python
        if _base_class_defined:
            base_field_names, class_vars, base_private_attributes = _collect_bases_data(bases)
            config_new = build_config(cls_name, bases, namespace, kwargs)
            ...
```

(Let me know if you want a different function name)